### ISSUE TITLE
KotshiPageJsonAdapter.java:25: error: illegal start of expression

### ISSUE THREAD

**Issue by blackberry2016** [ansman/kotshi#61]
Error Info:

> KotshiPageJsonAdapter.java:25: error: illegal start of expression
>     adapter0 = moshi.adapter(Types.newParameterizedType(List.class, ? extends T.class))

**adapter0 = moshi.adapter(Types.newParameterizedType(List.class, ? extends T.class));**

VO.kt
```java
@JsonSerializable
data class Page<T> constructor(@Json(name = "my_list") var list: List<T>)

@KotshiJsonAdapterFactory
object ApplicationJsonAdapterFactory : KotshiApplicationJsonAdapterFactory()
```

Generated KotshiPageJsonAdapter.java
```
public final class KotshiPageJsonAdapter<T> extends NamedJsonAdapter<Page<T>> {
  private static final JsonReader.Options OPTIONS = JsonReader.Options.of(
      "my_list");

  private final JsonAdapter<List<? extends T>> adapter0;

  public KotshiPageJsonAdapter(Moshi moshi, Type[] types) {
    super("KotshiJsonAdapter(Page)");
    adapter0 = moshi.adapter(Types.newParameterizedType(List.class, ? extends T.class));
  }

  @Override
  public void toJson(JsonWriter writer, Page<T> value) throws IOException {
    if (value == null) {
      writer.nullValue();
      return;
    }
    writer.beginObject();

    writer.name("my_list");
    adapter0.toJson(writer, value.getList());

    writer.endObject();
  }

  @Override
  public Page<T> fromJson(JsonReader reader) throws IOException {
    if (reader.peek() == JsonReader.Token.NULL) {
      return reader.nextNull();
    }

    reader.beginObject();

    List<? extends T> list = null;
    while (reader.hasNext()) {
      switch (reader.selectName(OPTIONS)) {
        case 0: {
          list = adapter0.fromJson(reader);
          continue;
        }
        case -1: {
          reader.nextName();
          reader.skipValue();
          continue;
        }
      }
    }

    reader.endObject();
    StringBuilder stringBuilder = null;
    if (list == null) {
      stringBuilder = KotshiUtils.appendNullableError(stringBuilder, "list");
    }
    if (stringBuilder != null) {
      throw new NullPointerException(stringBuilder.toString());
    }

    return new Page<T>(
        list);
  }
}

```

**Comment by kukuhyoniatmoko** [comment]
Maybe you need `@JvmSuppressWildcards`
```
@JsonSerializable
data class Page<T> constructor(@Json(name = "my_list") var list: List<@JvmSuppressWildcards  T>)
```

**Comment by ansman** [comment]
I implemented a fix in #64
